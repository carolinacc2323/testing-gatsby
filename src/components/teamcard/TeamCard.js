import React from 'react'
import './teamcard.css'
import gitlab from './gitlab.svg'
import linkedin from './linkedin.svg'


const TeamCard = ({ imageSrc, title, role, description }) => {
    return (
      <div className="card" data-testid="card-team">
        <img src={imageSrc} alt={title} className="card-image-team" />
        <div className='team-redes'>
          <img src={gitlab} width={20}/>
          <img src={linkedin} width={20}/>
        </div>
        <h2 className="card-title-team">{title}</h2>
        <h2 className="card-role-team">{role}</h2>
        <p className="card-description-team">{description}</p>
      </div>
    )
  }
  
  export default TeamCard
