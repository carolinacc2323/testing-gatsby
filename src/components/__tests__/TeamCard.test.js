import React from 'react';
import { render } from '@testing-library/react';
import TeamCard from '../teamcard/TeamCard';

describe('Card', () => {
  it('renderizamos bien el Componente', () => {
    const { container } = render(
    <TeamCard
        imageSrc="https://via.placeholder.com/150"
        title="Sample Title"
        role="jefe de equipo"
        description="This is a sample description for the card."
    />
    );
    expect(container.firstChild).toMatchSnapshot();
});
});
