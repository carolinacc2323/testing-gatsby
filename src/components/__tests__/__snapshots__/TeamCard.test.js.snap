// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renderizamos bien el Componente 1`] = `
<div
  class="card-team"
  data-testid="card-team"
>
  <img
    alt="Sample Title"
    class="card-team-image"
    src="https://via.placeholder.com/150"
  />
  <h2
    class="card-team-title"
  >
    Sample Title
  </h2>
  <h2
    class="card-team-role"
  >
    jefe de equipo
  </h2>
  <p
    class="card-team-description"
  >
    This is a sample description for the card.
  </p>
</div>
`;
